### PR TITLE
Alerting: Fix detail view not showing instances nor history tab

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes the alert detail view not showing the instances nor the history tab.

This is a regression after merging this [one](https://github.com/grafana/grafana/pull/92975)

**Why do we need this feature?**

These tabs should be shown.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:
Related to this [escalation](https://github.com/grafana/support-escalations/issues/12849)

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by gillesdemey — APPROVED** [review]
Code changes LGTM

File: public/app/features/alerting/unified/hooks/useCombinedRule.ts [hunk 1]
```diff
@@ -116,6 +118,12 @@ export function useCombinedRule({ ruleIdentifier, limitAlerts }: Props): Request
       refetchOnMountOrArgChange: true,
     }
   );
+  // in case of Grafana folder, we need to use the folder name instead of uid, as in promrules we don't use uid
+  const folder = useFolder(ruleLocation?.namespace);
+  const folderName =
+    ruleSourceName === GRAFANA_RULES_SOURCE_NAME && folder.folder
+      ? stringifyFolder(folder.folder)
+      : ruleLocation?.namespace;
```
Comment (konrad147) [inline public/app/features/alerting/unified/hooks/useCombinedRule.ts:126]: I think we can simplify this a bit
```suggestion
	const isGrafanaRule = isGrafanaRulesSource(ruleSourceName);
  const folder = useFolder(isGrafanaRule ? ruleLocation?.namespace : undefined);
  const namespaceName =  isGrafanaRule && folder.folder
      ? stringifyFolder(folder.folder)
      : ruleLocation?.namespace;
```

File: public/app/features/alerting/unified/hooks/useCombinedRule.ts [hunk 2]
```diff
@@ -141,15 +149,15 @@ export function useCombinedRule({ ruleIdentifier, limitAlerts }: Props): Request
 
     const rulerConfig = rulerRuleGroup ? { [ruleLocation.namespace]: [rulerRuleGroup] } : {};
```
Comment (konrad147) [inline public/app/features/alerting/unified/hooks/useCombinedRule.ts:150]:
```suggestion
    const rulerConfig = rulerRuleGroup ? { [namespaceName]: [rulerRuleGroup] } : {};
```

File: public/app/features/alerting/unified/hooks/useCombinedRule.ts [hunk 3]
```diff
@@ -141,15 +149,15 @@ export function useCombinedRule({ ruleIdentifier, limitAlerts }: Props): Request
...
 
-    const combinedNamespaces = combineRulesNamespaces(ruleSource, promRuleNs, rulerConfig);
+    const combinedNamespaces = combineRulesNamespace(ruleSource, promRuleNs, folderName ?? '', rulerConfig);
```
Comment (konrad147) [inline public/app/features/alerting/unified/hooks/useCombinedRule.ts:152]: If we apply the changes above then, the changes to this function should not be needed

**Review by konrad147 — APPROVED** [review]
LGTM!

**Pull request merged**